test/sync_fixtures.hpp

```diff
@@ -562,6 +567,9 @@ class MultiClientServerFixture {
     {
         unit_test::TestContext& test_context = m_test_context;
         stop();
+        for (int i = 0; i < m_num_clients; ++i) {
+            m_client_socket_providers[i]->stop(true);
```

Comment:
Shouldn't `stop()` do this (and remove it from here)?

Reply:
It doesn't hurt but not necessary, since the both the client (currently, but will be removed soon) and the DefaultWebsocketProvider does this in their destructors.

Reply:
And `stop()` doesn't need to do this, since `client.stop()` will already start the event loop stop process. But that will be changed with my upcoming changes - once those are in, it shouldn't matter if/when we call stop on the event loop. I can revisit with my changes.

Reply:
The client calls `stop()` but without `stop(true)`, so it doesn't really do this. Maybe we could replace this with `m_client_socket_providers.clear()` in the future, but since we had a bunch of weird data races I think it's worth it to try to keep this shutdown logic as close to the old behavior as possible.